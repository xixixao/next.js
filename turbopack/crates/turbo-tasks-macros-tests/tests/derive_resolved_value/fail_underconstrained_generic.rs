#![allow(dead_code)]

use turbo_tasks::ResolvedValue;

#[derive(ResolvedValue)]
struct ContainsUnderconstrainedGeneric<T> {
    value: T,
}

fn main() {}
