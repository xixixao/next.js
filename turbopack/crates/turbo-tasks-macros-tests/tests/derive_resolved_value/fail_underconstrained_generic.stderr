error[E0277]: the trait bound `T: ResolvedValue` is not satisfied
 --> tests/derive_resolved_value/fail_underconstrained_generic.rs:7:12
  |
7 |     value: T,
  |            ^ the trait `ResolvedValue` is not implemented for `T`
  |
note: required by a bound in `DeriveResolvedValueAssertion::<T>::assert_impl_resolved_value`
 --> tests/derive_resolved_value/fail_underconstrained_generic.rs:5:10
  |
5 | #[derive(ResolvedValue)]
  |          ^^^^^^^^^^^^^ required by this bound in `DeriveResolvedValueAssertion::<T>::assert_impl_resolved_value`
  = note: this error originates in the derive macro `ResolvedValue` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T`
  |
6 | struct ContainsUnderconstrainedGeneric<T: turbo_tasks::ResolvedValue> {
  |                                         ++++++++++++++++++++++++++++
